### ISSUE TITLE
refresh bundles automatically

### ISSUE THREAD

**Issue by percyliang** [codalab/codalab-worksheets#104]
For running bundles on a worksheet that are rendered as tables (and eventually graphs, etc.), the react element should update automatically without reloading the entire worksheet.  The contents of the side panel should also refresh (including stdout, stderr).

We might want to batch bundles if this is easy to minimize the number of rest calls.

We should decide on a reasonable refresh frequency - maybe once per second.

**Comment by kashizui** [comment]
@percyliang @ChengshuLi change has been deployed to wstest for QA, please play around with it.

**Comment by percyliang** [comment]
@ChengshuLi Something's pretty buggy...try typing 'run date' 10 times
really quickly at the web terminal.  First, less than 10 rows show up
(usually ~7), and refreshing kind of get stalled.  Can you reproduce / fix
this before we merge develop into master?

**Comment by ChengshuLi** [comment]
Yes will do tonight and let you know!

**Comment by ChengshuLi** [comment]
@percyliang 

I wasn't able to reproduce this bug after trying it on different browsers. Just want to clarify what you mean by "try typing 'run date' 10 times really quickly at the web terminal."

What I did is to 'run date', and then press "up arrow" and "enter" to re-run 'run date'. I did this for 20 times. The refreshing seems fine. All bundles finish to 'ready' state. One thing that I did notice is that I couldn't do this ("up arrow" and "enter") very quickly. The web terminal took half a second to react. 

For example: 
`CodaLab> cl run 'date';`
`0x78715702c84f496eb74d7f66136844a0`

Basically I need to wait for the bundle uuid to be printed out, before I run the next 'run date'. Otherwise when I press "up arrow", I won't see 'run date', which is the previous command. I only see blank, although I guess it's probably fine for a general user?

**Comment by percyliang** [comment]
Yes, I was also pressing 'up' and 'enter' multiple times, pressing up as
soon as the uuid prints.  Maybe this happens more with a slower Internet
connection?  I'm on Linux Chrome.

When something goes wrong, I get this in the JS console:
test.codalab.org/rest/api/bundles/0xd2ba4c3ea920420c85598ab38350b9b2/?_=1471843652703
Failed to load resource: the server responded with a status of 404 (Not
Found)
I'm guessing this is some weird timing thing where the bundle hasn't been
created properly when it is being loaded...

 -Percy

**Comment by ChengshuLi** [comment]
@percyliang  hrm... that's bizarre. I agree with your reasoning that it's probably because the bundle hasn't been created properly when it is being loaded. Could this be a backend problem? Like the front end's request is racing against the backend trying to create the bundle? I am not sure.

Trying to see whether @kashizui could reproduce this bug.

**Comment by kashizui** [comment]
Here is my reproduction of the bug on Chrome Version 52.0.2743.116 (64-bit), OS X 10.11.6:

http://web.stanford.edu/~sckoo/images/bugdemo.mov

**Comment by kashizui** [comment]
This feature has been implemented and is in production, but there are some issues tracked by #211
